src/algebraic_geometry/Gamma_Spec_adjunction.lean

```diff
@@ -0,0 +1,370 @@
+one of the two categories to have arrows reversed, and it is equally valid to present
+the adjunction as `Spec ⊣ Γ` (`Spec.to_LocallyRingedSpace.right_op ⊣ Γ`), in which
+case the unit and the counit would switch to each other.
+
+## Main definition
+
+* `algebraic_geometry.identity_to_Γ_Spec` : The natural transformation `𝟭 _ ⟶ Γ ⋙ Spec`.
+* `algebraic_geometry.Γ_Spec.LocallyRingedSpace_adjunction` : The adjunction `Γ ⊣ Spec` from
+  `CommRingᵒᵖ` to `LocallyRingedSpace`.
+* `algebraic_geometry.Γ_Spec.adjunction` : The adjunction `Γ ⊣ Spec` from
+  `CommRingᵒᵖ` to `Scheme`.
+
+-/
+
+noncomputable theory
+universes u
+
+open prime_spectrum
+
+namespace algebraic_geometry
+open opposite
+open category_theory
+open structure_sheaf
+open topological_space
+open algebraic_geometry.LocallyRingedSpace
+open Top.presheaf
+open Top.presheaf.sheaf_condition
+
+namespace LocallyRingedSpace
+
+variable (X : LocallyRingedSpace.{u})
+
+/-- Map from the global sections to a stalk. -/
+def Γ_to_stalk (x : X) : Γ.obj (op X) ⟶ X.presheaf.stalk x :=
+  X.presheaf.germ (⟨x,trivial⟩ : (⊤ : opens X))
+-- or @Top.presheaf.germ _ _ _ _ _ ⊤ ⟨x,trivial⟩
+
+/-- Unit on the underlying set. -/
+def to_Γ_Spec_fun : X → prime_spectrum (Γ.obj (op X)) :=
+  λ x, comap (X.Γ_to_stalk x) (@local_ring.closed_point _ _ (X.local_ring x))
+-- or Spec.to_Top.map (X.Γ_to_stalk x).op (@local_ring.closed_point ...)
+
+lemma not_mem_prime_iff_unit_in_stalk (r : Γ.obj (op X)) (x : X) :
+  r ∉ (X.to_Γ_Spec_fun x).as_ideal ↔ is_unit (X.Γ_to_stalk x r) :=
+by erw [local_ring.mem_maximal_ideal, not_not]
+
+/-- Preimage of a basic open in `Spec Γ(X)` under the unit is the basic open in `X`
+    defined by the same element (equal as sets). -/
+lemma to_Γ_Spec_preim_basic_open_eq (r : Γ.obj (op X)) :
+  X.to_Γ_Spec_fun⁻¹' (basic_open r).1 = (X.to_RingedSpace.basic_open r).1 :=
+by { ext, erw X.to_RingedSpace.mem_top_basic_open, apply not_mem_prime_iff_unit_in_stalk }
+
+/-- Unit is continuous. -/
+lemma to_Γ_Spec_continuous : continuous X.to_Γ_Spec_fun :=
+begin
+  apply is_topological_basis_basic_opens.continuous,
+  rintro _ ⟨r, rfl⟩,
+  erw X.to_Γ_Spec_preim_basic_open_eq r,
+  exact (X.to_RingedSpace.basic_open r).2,
+end
+
+/-- Unit as a continuous map. -/
+@[simps]
+def to_Γ_Spec_base : X.to_Top ⟶ Spec.Top_obj (Γ.obj (op X)) :=
+{ to_fun := X.to_Γ_Spec_fun,
+  continuous_to_fun := X.to_Γ_Spec_continuous }
+
+variable (r : Γ.obj (op X))
+
+/-- The preimage in `X` of a basic open in `Spec Γ(X)` (as an open set). -/
+abbreviation to_Γ_Spec_map_basic_open : opens X :=
+(opens.map X.to_Γ_Spec_base).obj (basic_open r)
+
+/-- The preimage is the basic open in `X` defined by the same element `r`. -/
+lemma to_Γ_Spec_map_basic_open_eq : X.to_Γ_Spec_map_basic_open r = X.to_RingedSpace.basic_open r :=
+subtype.eq (X.to_Γ_Spec_preim_basic_open_eq r)
+
+/-- The map from the global sections `Γ(X)` to the sections on the (preimage of) a basic open. -/
+abbreviation to_to_Γ_Spec_map_basic_open :
```

Comment:
The `to_to_` isn't very great. But maybe this should decl should be considered an implementation detail never to be seen again outside this file?

Reply:
I don't think this will be referred anywhere outside this file.